api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -0,0 +1,530 @@
+	// completes. After the TTL expires, the resource will be automatically
+	// deleted without the user having to take any direct action.
+	//
+	// If this field is unset then the request resource will not be
+	// automatically deleted. If this field is set to zero then the request
+	// resource is eligible for deletion immediately after it finishes.
+	//
+	// +optional
+	TTLSecondsAfterFinished *int64 `json:"ttlSecondsAfterFinished,omitempty"`
+}
+
+// VirtualMachinePublishRequestStatus defines the observed state of a
+//  VirtualMachinePublishRequest.
+type VirtualMachinePublishRequestStatus struct {
+	// CompletionTime represents time when the request was completed. It is not
+	// guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	//
+	// The value of this field should be equal to the value of the
+	// LastTransitionTime for the status condition Type=Complete.
+	//
+	// +optional
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+
+	// StartTime represents time when the request was acknowledged by the
+	// controller. It is not guaranteed to be set in happens-before order
+	// across separate operations. It is represented in RFC3339 form and is
+	// in UTC.
+	//
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// ImageName is the name of the VirtualMachineImage resource that is
+	// eventually realized in the same namespace as the VM and publication
+	// request after the publication operation completes.
+	//
+	// This field will not be set until the VirtualMachineImage resource
+	// is realized.
+	//
+	// +optional
+	ImageName string `json:"imageName,omitempty"`
+
+	// Ready is set to true only when the VM has been published successfully
+	// and the new VirtualMachineImage resource is ready.
+	//
+	// Readiness is determined by waiting until there is status condition
+	// Type=Complete and ensuring it and all other status conditions present
+	// have a Status=True. The conditions present will be:
+	//
+	//   * SourceValid
+	//   * TargetValid
+	//   * Uploaded
+	//   * ImageAvailable
+	//   * Complete
+	//
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// Conditions is a list of the latest, available observations of the
+	// request's current state.
+	//
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+func (vmpr VirtualMachinePublishRequest) getCondition(
+	conditionType string,
+) *metav1.Condition {
+
+	for i := range vmpr.Status.Conditions {
+		c := &vmpr.Status.Conditions[i]
+		if c.Type == conditionType {
+			return c
+		}
+	}
+
+	return nil
+}
+
+// IsSourceValid returns true if there is a status condition of Type=SourceValid
+// and Status=True.
+func (vmpr VirtualMachinePublishRequest) IsSourceValid() bool {
+	c := vmpr.getCondition(VirtualMachinePublishRequestConditionSourceValid)
+	if c != nil && c.Status == metav1.ConditionTrue {
+		return true
+	}
+	return false
```

Comment:
Perhaps we can simply have `return c != nil && c.Status == metav1.ConditionTrue` here and below as well?